### PR TITLE
Implement time:* 2.0.0 & 3.0.0 commands

### LINKED ISSUE #682: timesrv IStaticService: 200 (IsStandardNetworkSystemClockAccuracySufficient) not implemented
timesrv IStaticService: 200 (IsStandardNetworkSystemClockAccuracySufficient) is not implemented.

## Service description:
[Reswitched](https://reswitched.github.io/SwIPC/ifaces.html#nn::timesrv::detail::service::IStaticService(200))
```
IsStandardNetworkSystemClockAccuracySufficient() -> b8
```

## Required by:
* https://github.com/Ryujinx/Ryujinx-Games-List/issues/102

### PR DESCRIPTION
**Pull request by Thog**
Also implement bpc:r as a result of 2.0.0 changes.

Close #682

### CODE REVIEW

File: Ryujinx.HLE/HOS/Services/Time/Clock/ClockTypes.cs [hunk 1]
```diff
@@ -1,40 +1,62 @@
...
-            return new TimeSpanType(ticks * 1000000000 / frequency);
+            return new TimeSpanType((long)ticks * 1000000000 / (long)frequency);
         }
     }
 
     [StructLayout(LayoutKind.Sequential)]
     struct SteadyClockTimePoint
     {
-        public ulong   TimePoint;
+        public long   TimePoint;
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Time/Clock/ClockTypes.cs]: nit: Alignment.

File: Ryujinx.HLE/HOS/Services/Bpc/IRtcManager.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+namespace Ryujinx.HLE.HOS.Services.Bpc
+{
+    [Service("bpc:r")]
+    class IRtcManager : IpcService
+    {
+        public IRtcManager(ServiceCtx context) { }
+
+        [Command(0)]
+        // GetRtcTime() -> u64
+        public static ResultCode GetRtcTime(ServiceCtx context)
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Bpc/IRtcManager.cs]: I think this one should be a instance method, even if theres no internal state that needs to be accessed, for consistency.

File: Ryujinx.HLE/HOS/Services/Bpc/IRtcManager.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                context.ResponseData.Write(rtcValue);
+            }
+
+            return result;
+        }
+
+        public static ResultCode GetExternalRtcValue(out ulong rtcValue)
+        {
+            // TODO: emulate MAX77620/MAX77812
+            DateTime UnixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Services/Bpc/IRtcManager.cs]: `unixEpoch`.

File: Ryujinx.HLE/HOS/Services/Time/Clock/StandardNetworkSystemClockCore.cs [hunk 4]
```diff
@@ -28,6 +30,7 @@ public StandardNetworkSystemClockCore(SteadyClockCore steadyClockCore)
             _context         = new SystemClockContext();
 
             _context.SteadyTimePoint.ClockSourceId = steadyClockCore.GetClockSourceId();
+            _standardNetworkClockSufficientAccuracy = new TimeSpanType(0);
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Time/Clock/StandardNetworkSystemClockCore.cs]: nit: Alignment.

File: Ryujinx.HLE/HOS/Services/Time/Clock/StandardNetworkSystemClockCore.cs [hunk 5]
```diff
@@ -55,5 +58,26 @@ public override ResultCode SetSystemClockContext(SystemClockContext context)
 
             return ResultCode.Success;
         }
+
+        public bool IsStandardNetworkSystemClockAccuracySufficient(KThread thread)
+        {
+            SteadyClockCore steadyClockCore = GetSteadyClockCore();
```
Comment (AcK77) [inline Ryujinx.HLE/HOS/Services/Time/Clock/StandardNetworkSystemClockCore.cs]:
```suggestion
            SteadyClockCore      steadyClockCore  = GetSteadyClockCore();
            SteadyClockTimePoint currentTimePoint = steadyClockCore.GetCurrentTimePoint(thread);
```

**Pull request merged**